pallets/parachain-staking/src/lib.rs

```diff
@@ -2042,53 +2058,25 @@ pub mod pallet {
 			log::trace!("{} Candidates for {} Collator seats", candidates.len(), top_n);
 
 			// Order candidates by their total stake (greatest to least)
-			candidates.sort_by(|a, b| b.amount.cmp(&a.amount));
+			//
+			// NOTE: Resorting is required if a collator's total stake just changed before
+			// updating the selected candidates.
+			candidates.sort_by(|a, b| b.cmp(&a));
```

Comment:
Ought to work:

```suggestion
			candidates.sort();
```

Reply:
This does not yield the reverse sorting that we require, e.g., it sorts from lowest to greatest.

Reply:
Didn't see that it's a reverse sort.